src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/MemberAndDeclarationInfo.cs

```diff
@@ -245,8 +245,47 @@ private sealed class MemberNameComparer : IComparer<MemberAndDeclarationInfo>
         {
             public int Compare(MemberAndDeclarationInfo x, MemberAndDeclarationInfo y)
             {
-                var comp = string.Compare(x.Name, y.Name, StringComparison.Ordinal);
-                return comp != 0 ? comp : (y._inheritanceLevel - x._inheritanceLevel);
+                var xName = x.Name;
+                var yName = y.Name;
+                var comp = string.Compare(xName, yName, StringComparison.Ordinal);
+                if (comp == 0)
+                {
+                    return y._inheritanceLevel - x._inheritanceLevel;
+                }
+
+                // Members with underscore prefix come first
+                var xUnderscore = xName.Length > 0 && xName[0] == '_';
+                var yUnderscore = yName.Length > 0 && yName[0] == '_';
+                if (xUnderscore && yUnderscore)
+                {
+                    return comp;
+                }
+                if (xUnderscore)
+                {
+                    return -1;
+                }
+                if (yUnderscore)
+                {
+                    return 1;
+                }
+
+                // Members with letter+underscore prefix (ie. m_member) come second
+                var xLetterUnderscore = xName.Length > 1 && xName[1] == '_';
+                var yLetterUnderscore = yName.Length > 1 && yName[1] == '_';
```

Comment:
This does mean that `s_` variables will come before regular non-prefixed instance members. Is that what we want?

Reply:
Yes, two naming conventions dominate for private fields: `_` prefix and hungarian.
This PR brings both to the top of the list in a way that will still allow users to find the item they are looking for.